doc/cpp/probe_sample.rst

```diff
@@ -387,16 +386,16 @@ as a way of passing additional metadata about a probe to any sampler
 that polls it, with a view to samplers that handle multiple probes,
 possibly with different value types.
 
-Probe addresses are decoupled from the cell descriptions themselves —
+probeset addresses are decoupled from the cell descriptions themselves —
 this allows a recipe implementation to construct probes independently
 of the cells themselves. It is the responsibility of a cell group implementation
-to parse the probe address objects wrapped in the ``any address`` field.
+to parse the probeset address objects wrapped in the ``any address`` field.
 
 The _k_th element of the vector returned by ``get_probes(gid)`` is
```

Comment:
Note: this is a strange choice. It's very brittle; imagine a user constructing the vector from an unordered container.

Reply:
I believe some chatter on Gitter elaborated on this. Should I add some of that here?